packages/web-components/fast-foundation/src/spin-button/spin-button.spec.md

```diff
@@ -0,0 +1,101 @@
+# Spin button
+
+## Overview
+
+A spin button component is the combination of two buttons and a text field that allows user to increment or decrement numeric value set by specific range or `Infinity` by default. It also allows the user to perform the same through the use of an up and down buttons on the keyboard. As defined by the W3C:
+
+> A spinbutton is an input widget that restricts its value to a set or range of discrete values.
+
+### Use Cases
+
+-   Alarm, Date, Clock, Persons' age and many different applications.
+
+### Features
+
+-   `min` and `max` attributes allow to set the range
+-   `step` attribute to define the difference on increment or decrement of the value
+-   `ArrowUp`/`Home` and `ArrowDown`/`End` keys to increment or decrement the value respectively
+-   In the mobiles/tablets, on focus it pops up native numeric keyboard
+-   Focus Delegation
+
+### Prior Art/Examples
+
+-   [BootstrapVue](https://bootstrap-vue.org/docs/components/form-spinbutton)
+-   [Office Fabric](https://developer.microsoft.com/en-us/fluentui#/controls/web/spinbutton)
+-   [Ant Design](https://ant.design/components/input-number/)
+-   [Shopify Polaris](https://polaris.shopify.com/components/forms/stepper)
+-   [React Suite](https://rsuitejs.com/components/input-number)
+
+---
+
+### API
+
+-   _Component Name:_ `fast-spin-button`
+-   _Props/Attrs:_
+    -   `min: number` - range starts from
+    -   `max: number` - range ends to
+    -   `step: number` - difference on each increment or decrement
+    -   `value: number` - default value
+    -   `autofocus: boolean` - automatically focuses the control
+    -   `placeholder: string` - an exemplar value to display in the input field whenever it is empty
+    -   `required: boolean` - boolean value that sets the field as required
+    -   `disabled: boolean` - disables the control
+    -   `readonly` - the text field should be submitted with the form but should not be editable.
+-   _Events_
+    -   `change: CustomEvent` - No custom data.
+
```

Comment:
What were you thinking the `parts` would be for this component?
This will allow "parts" of the element shadowDom (below) to be exposed and easy to access mostly for style changes.
https://www.w3.org/TR/css-shadow-parts-1/#:~:text=This%20specification%20defines%20the%20%3A%3Apart%20%28%29%20pseudo-element%20on,XML%29%20on%20screen%2C%20on%20paper%2C%20in%20speech%2C%20etc.

We have plenty of other specs in our repo that you can refer to to get an idea :)